ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -272,21 +272,8 @@ def with_metadata(metadata_for_block, &block)
     # {http://railseventstore.org/docs/subscribe/#async-handlers Read more}
     #
     # @return [Event] deserialized event
-    def deserialize(serializer:, event_type:, event_id:, data:, metadata:, timestamp: nil, valid_at: nil)
-      extract_timestamp = lambda { |m| (m[:timestamp] || Time.parse(m.fetch("timestamp"))).iso8601 }
-
-      mapper.record_to_event(
-        SerializedRecord
-          .new(
-            event_type: event_type,
-            event_id: event_id,
-            data: data,
-            metadata: metadata,
-            timestamp: timestamp || timestamp_ = extract_timestamp[serializer.load(metadata)],
-            valid_at: valid_at || timestamp_
-          )
-          .deserialize(serializer)
-      )
+    def deserialize(serializer:, **keywords)
+      _deserialize(serializer: serializer, **keywords.transform_keys(&:to_sym))
```

Comment:
👎 it is expected that an upper layer passes expected keyword arguments — no need to change this method